test_runner/regress/test_tenants.py

```diff
@@ -475,3 +476,37 @@ def only_int(samples: List[Sample]) -> int:
     assert counts
     log.info(f"directory counts: {counts}")
     assert counts[2] > COUNT_AT_LEAST_EXPECTED
+
+
+def test_tenants_parallel_creation(neon_simple_env: NeonEnv):
+    """
+    (Relaxed) regression test for issue that led to https://github.com/neondatabase/neon/pull/9268
+    Create many tenants in parallel and then restart endpoints
+    """
+    env = neon_simple_env
+    pageserver_http_client = env.pageserver.http_client()
+
+    # This param needs to be 200 to reproduce the limit issue
+    n_threads = 8
+    barrier = threading.Barrier(n_threads)
+
+    def test_tenant():
+        # Create new tenant
+        tenant_id, _ = env.create_tenant()
+        endpoint = env.endpoints.create_start("main", tenant_id=tenant_id)
+        time.sleep(2)
+        endpoint.stop()
+        time.sleep(2)
+        # Use a barrier to make sure we restart endpoints at the same time
+        barrier.wait()
+        endpoint.start()
```

Comment:
is the idea that restarting an endpoint causes a broker connection being made?

Reply:
If that is the case, I don't think it makes sense: when the timeline is started the walreceiver is started thus the connection is made. I cannot see how this could make the case more apparent.

Reply:
that is not enough to trigger the issue: just creation alone doesn't (reliably) reproduce errors.